glastopf/modules/reporting/auxiliary/log_s3.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (C) 2018 Andre Vorbach @vorband
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import os
+import gevent
+
+import botocore.session, botocore.client
+from botocore.exceptions import ClientError
+
+from glastopf.modules.reporting.auxiliary.base_logger import BaseLogger
+
+
+logger = logging.getLogger(__name__)
+
+
+class S3Logger(BaseLogger):
+
+    def __init__(self, data_dir, work_dir, config="glastopf.cfg", reconnect=True):
+        config = os.path.join(work_dir, config)
+        BaseLogger.__init__(self, config)
+        self.files_dir = os.path.join(data_dir, 'files/')
+        self.enabled = False
+        self._initial_connection_happend = False
+        self.options = {'enabled': self.enabled}
+        if self.config.getboolean("s3storage", "enabled"):
+            self.endpoint = self.config.get("s3storage", "endpoint")
+            self.accesskey = self.config.get("s3storage", "aws_access_key_id")
+            self.secretkey = self.config.get("s3storage", "aws_secret_access_key")
+            self.version = self.config.get("s3storage", "signature_version")
+            self.region = self.config.get("s3storage", "region")
+            self.bucket = self.config.get("s3storage", "bucket")
+            self.enabled = True
+            self.options = {'enabled': self.enabled}
+            self.s3client = None
+            self.s3session = None
+            gevent.spawn(self._start_connection, self.endpoint, self.accesskey, self.secretkey, self.version, self.region, self.bucket)
```

Comment:
maybe make this blocking so we ensure we have a connection

Reply:
I added a test if bucket is available and if it fails during initial start, it will no longer try.
Will update the module now.